PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRunner.java

```diff
@@ -122,6 +124,17 @@ public ISparkBatchJob buildSparkBatchJob(@NotNull SparkSubmitModel submitModel,
                 try {
                     clusterDetail.getConfigurationInfo();
                     storageAccount = clusterDetail.getStorageAccount();
+
+                    if (storageAccount.getAccountType() == StorageAccountTypeEnum.ADLSGen2) {
+                        destinationRootPath = submitModel.getJobUploadStorageModel().getUploadPath();
+                        accessKey = ((ADLSGen2StorageAccount) storageAccount).getPrimaryKey();
+                        if (StringUtils.isBlank(accessKey)) {
+                            throw new ExecutionException("Cannot get valid access key for storage account");
+                        }
+
+                        httpObservable = new SharedKeyHttpObservable(storageAccount.getName(), accessKey);
```

Comment:
Could you please create an issue to fix it with `ClusterDetail`'s shared `httpObservable`?